### ISSUE TITLE
Bug 58070094: [React Native Gallery -> Left Navigation]: Keyboard focus and Narrator focus are not on sync when user tries to navigate to the combo boxes present in left navigation pane.

### ISSUE THREAD

**Issue by vineethkuttan** [microsoft/react-native-gallery#611]
Build Details:
OS Version: Dev (OS Build 27871.1000)
App: React Native Gallery 
App Version: 1.0.18.0
React Native Windows Version: 0.79.0

AT: Screen Reader

Pre-requisites:

Turn on Narrator (Ctrl + Win + Enter).

Repro-Steps:
Launch the React Native Gallery app.
Navigate to ‘Expand menu’ button and activate it.
Now try to navigate in between in the combo boxes and access those.
Actual Result:
Keyboard focus and Narrator focus are not on sync when user tries to navigate to the combo boxes present in left navigation pane. When user hit tab key, keyboard focus is falling on the combo boxes but narrator focus is not in with sync with keyboard focus as narrator focus is getting stuck in 'All Samples' button and narrator stays on mute.

Expected Result:
Keyboard focus and Narrator focus should be on sync when user tries to navigate to the combo boxes present in left navigation pane. When keyboard focus fall on combo boxes, narrator focus should also fall on the combo boxes along with the keyboard focus and narrator should announce about it. 

User Impact:
Users with visual impairment who rely on screen reader will get confused if narrator focus and Keyboard focus are not sync.

MAS Reference:
[MAS 4.3.1 – No Disruption of Accessibility Features](https://aka.ms/MAS4.3.1)

Trap ID:
[1.4 - Uncomprehended Element](https://aka.ms/UncomprehendedElement)

"Have feedback on bugs logged, please tag bug as “A11yRCA” and add your feedback in the comment section.”

**Comment by chiaramooney** [comment]
This should only be a JS side change (no Paper architecture specific knowledge needed). Seems like the nav menu focus code needs to be looked at. 

Note: The source code for the Paper React Native Gallery begins at the root of the repo rather than inside the New Arch folder.

**Comment by protikbiswas100** [comment]
We observed same issue for both paper and fabric. @vineethkuttan to check with accessibility team about this issue

**Comment by protikbiswas100** [comment]
Raised pr https://github.com/microsoft/react-native-gallery/pull/644
